spec/requests/subjects_request_spec.rb

```diff
@@ -0,0 +1,139 @@
+require 'rails_helper'
+
+RSpec.describe 'Admin::Subjects', type: :request do
+  let(:subject) { create(:subject) }
+  let(:subjects) { create_list(:subject, 3) }
+  let!(:admin) { create(:user,
+                        admin: true,
+                        address: "Kharkov",
+                        phone: '+38(023)122-2222',
+                        first_name: "Zakhar",
+                        last_name: 'Bondar') }
+  let(:attr) { { name: 'Physics' } }
+  let(:wrong_attr) { { name: '' } }
+  let(:invalid_id) { '998' }
+
+  describe 'GET #index' do
+    before(:each) do
+      sign_in admin
```

Comment:
add context with logged-in admin and use this there

Reply:
Addded.